main.cpp

```diff
@@ -1,25 +1,95 @@
 #include "main.h"
 #include "rc.h"
+#include <stdexcept>
+
+using namespace std;
 
 int number_dialog::idd() const {
 	return IDD_NUMBER; 
 }
 bool number_dialog::on_init_dialog(){
+	this->set_int(IDC_EDIT1, range);
 	return true;
 }
 bool number_dialog::on_ok(){
+	try {
+		range = this->get_real(IDC_EDIT1);
+	}
+	catch(std::runtime_error err) {
+		const char * what = err.what();
+		std::wstring msg(what, what + strlen(what));
+		MessageBox(*this, msg.c_str(), _T("Error"), MB_OK | MB_ICONERROR);
+	}
+	
 	return true;
 }
  
 void main_window::on_paint(HDC hdc){
+	int range = nd.get_range();
+	if (range <= 0) {
+		return;
+	}
+
+	RECT parent_rect;
+	GetClientRect(*this, &parent_rect);
+
+	int height = parent_rect.bottom / (range + 1);
+	int width = parent_rect.right / (range + 1);
+
+	MoveToEx(hdc, 0, height, 0);
+	LineTo(hdc, parent_rect.right, height);
+	MoveToEx(hdc, width, 0, 0);
+	LineTo(hdc, width, parent_rect.bottom);
+
+	SetTextColor(hdc, this->font_color);
+
+	auto old = SelectObject(hdc, this->font);
+
+	for (int i = 0; i <= range; i++){
+		for (int j = 0; j <= range; j++) {
+			RECT r = { width * j, height * i, (width * j) + width, (height * i) + height };
+			std::wstring wstr = (i == 0 || j == 0) ? std::to_wstring(i | j) : std::to_wstring(i * j);

```

Comment:
Simpatično, može još kraće (ako vam je to čitko):
	`const auto s = std::to_wstring(i && j ? i * j : i | j);`